src/Resources/ResourceManager/Implementation/CmdletBase/ResourceManagerCmdletBase.cs

```diff
@@ -423,7 +423,8 @@ private void HandleException(ExceptionDispatchInfo capturedException)
                 var errorResponseException = capturedException.SourceException as ErrorResponseMessageException;
                 if (errorResponseException != null)
                 {
-                    this.ThrowTerminatingError(errorResponseException.ToErrorRecord());
+                    this.WriteError(errorResponseException.ToErrorRecord());
```

Comment:
So, there are cases in which we will want to throw a terminating exception (such as PipelineStoppedException)

If this is extending AzurePSCmdlet and usign the default ProcessRecord implementation, we can make use of it's default error handling, and simply throw these exceptions (it will write them to the error stream): https://github.com/Azure/azure-powershell-common/blob/master/src/Common/AzurePSCmdlet.cs#L735.

Otherwise, we want to throw any exception that is a terminating exception and write others to the error stream. We can use the inherited IsTerminatingException method to do this.